vars/pythonSetupPyPipeline.groovy

```diff
@@ -24,23 +24,35 @@ def call(Map pipelineParams) {
     pipelineParams["dockerFilename"] = "Dockerfile"
   }
   if (!pipelineParams.dockerBuildArgs) {
-    pipelineParams["dockerBuildArgs"] = ""
+    pipelineParams["dockerBuildArgs"] = "--no-cache --network host"
   }
   if (!pipelineParams.dockerRunArgs) {
     pipelineParams["dockerRunArgs"] = "-v /etc/passwd:/etc/passwd:ro -v /opt/jenkins/.ssh:/opt/jenkins/.ssh:ro --network host"
   }
 
-  pipeline {
-    agent {
-      dockerfile {
-        filename pipelineParams.dockerFilename
-        additionalBuildArgs pipelineParams.dockerBuildArgs
-        args pipelineParams.dockerRunArgs
-      }
+  if (pipelineParams.dockerDeploy) {
+    if (!pipelineParams.dockerRegistryUrl) {
+      error("${LOG_TAG} Please provide a Docker registry URL (eg. https://registry.example.com)")
+    }
+    if (!pipelineParams.dockerRegistryCredentialsId) {
+      error("${LOG_TAG} Please provide a Jenkins credentials id for deploying to ${pipelineParams.dockerRegistryUrl}")
+    }
+    if (!pipelineParams.dockerRepo) {
+      error("${LOG_TAG} Please provide a Docker repository name (eg. acme). The image name will be based on the Python module name (eg. acme/mymodule)")
     }
+    if (!pipelineParams.dockerDeployFile) {
+      echo("${LOG_TAG} Using default Dockerfile ('Dockerfile.deploy') for deployment")
+      pipelineParams["dockerDeployFile"] = "Dockerfile.deploy"
```

Comment:
I only see Dockerfile.build in the cookiecutter template at the moment, will this be added there?

Reply:
Yes, that will be as well available in the template